src/Illuminate/Foundation/Http/Middleware/TrimStrings.php

```diff
@@ -62,13 +62,21 @@ protected function transform($key, $value)
     {
         $except = array_merge($this->except, static::$neverTrim);
 
-        if (in_array($key, $except, true) || ! is_string($value)) {
+        if ($this->skipTrim($key, $except) || ! is_string($value)) {
```

Comment:
You've just moved `in_array` to a function, just this!!

Reply:
@milwad-dev 
Yes, this is a hook point. You can customize the behavior like this:

```php
class TrimArrayStrings extends TrimStrings
{
  protected function skipTrim($key, $except)
  {
    return parent::skipTrim($key, $except) || $this->skipMoreComplexCondition($key, $except) 
  }
}

```